src/tests/GC/Stress/Framework/ReliabilityFramework.cs

```diff
@@ -1081,20 +1070,42 @@ private void TestSetShutdown(int totalTestsToRun)
             _logger.WriteToInstrumentationLog(_curTestSet, LoggingLevels.StartupShutdown, msg);
             Console.WriteLine(msg);
 
+            int secondsIter = 5;
             int waitCnt = 0;
-            while (_testsRunningCount > 0 && waitCnt < 7200)  // wait a max of 2 hours for the tests to exit...
-            {
-                Thread.Sleep(1000);
-                Console.Write("\b\b\b\b{0,4}", _testsRunningCount);
+            int waitCntTotal = _curTestSet.MaximumWaitTime * 60 / secondsIter;
+            msg = String.Format("START WAITING for {0}s", _curTestSet.MaximumWaitTime * 60);
+            _logger.WriteToInstrumentationLog(_curTestSet, LoggingLevels.StartupShutdown, msg);
+            long lastAllocatedBytes = GC.GetTotalAllocatedBytes(false);
+            while (_testsRunningCount > 0 && waitCnt < waitCntTotal)
+            {
+                Thread.Sleep(secondsIter * 1000);
+                long currentAllocatedBytes = GC.GetTotalAllocatedBytes(false);
+                msg = String.Format("============current number of tests running {0,4}, allocated {1:n0} so far, {2:n0} since last; (GC {3}:{4}:{5}), waited {6}s",
+                    _testsRunningCount, currentAllocatedBytes, (currentAllocatedBytes - lastAllocatedBytes),
+                    GC.CollectionCount(0),
+                    GC.CollectionCount(2),
+                    GC.CollectionCount(2),
+                    (waitCnt * secondsIter));
+                lastAllocatedBytes = currentAllocatedBytes;
+                _logger.WriteToInstrumentationLog(_curTestSet, LoggingLevels.StartupShutdown, msg);
+
+                for (int i = 0; i < _curTestSet.Tests.Length; i++)
+                {
+                    if (_curTestSet.Tests[i].RunningCount != 0)
+                    {
+                        msg = String.Format("Still running: {0}", _curTestSet.Tests[i].RefOrID);
+                        _logger.WriteToInstrumentationLog(_curTestSet, LoggingLevels.StartupShutdown, msg);
+                    }
+                }
                 waitCnt++;
             }
         }
 
         // let the user know what tests haven't finished...
         if (_testsRunningCount != 0)
         {
-            string msg;
-
+            string msg = String.Format("************Timeout reached************");
```

Comment:
The call to `String.Format` is unnecessary?

Reply:
right 😁 that's from some copying/pasting. will get rid of it